pkg/operator/controllers/pullsecret/pullsecret_controller.go

```diff
@@ -193,3 +239,106 @@ func (r *PullSecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Named(controllers.PullSecretControllerName).
 		Complete(r)
 }
+
+func (r *PullSecretReconciler) parseRHRegistryKeys(ps *v1.Secret) (*serializedAuthMap, error) {
+	var pullSecretData *serializedAuthMap
+	if data := ps.Data[v1.DockerConfigJsonKey]; len(data) > 0 {
+		if err := json.Unmarshal(data, &pullSecretData); err != nil {
+			return nil, err
+		}
+	}
+	return pullSecretData, nil
+}
+
+// checkRHRegistryKeys checks whether the rhRegistry keys:
+//   - redhat.registry.io
+//   - registry.connect.redhat.com"
+// are present in the pullSecret
+func (r *PullSecretReconciler) checkRHRegistryKeys(psData *serializedAuthMap) (foundKeys []string) {
+	rhKeys := []string{
+		"registry.redhat.io",
+		"registry.connect.redhat.com",
```

Comment:
I dont think we care about this key. Samples operator does not operate with this secret. So would be good to simplify on the one key, we are interested in.

Reply:
`"registry.connect.redhat.com",` I ref to this